pkg/controller/pxcbackup/deadline.go

```diff
@@ -0,0 +1,104 @@
+package pxcbackup
+
+import (
+	"context"
+	"time"
+
+	"github.com/pkg/errors"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	api "github.com/percona/percona-xtradb-cluster-operator/pkg/apis/pxc/v1"
+)
+
+var (
+	errSuspendedDeadlineExceeded = errors.New("suspended deadline seconds exceeded")
+	errStartingDeadlineExceeded  = errors.New("starting deadline seconds exceeded")
+)
+
+func (r *ReconcilePerconaXtraDBClusterBackup) checkDeadlines(ctx context.Context, cluster *api.PerconaXtraDBCluster, cr *api.PerconaXtraDBClusterBackup) error {
+	if err := checkStartingDeadline(ctx, cluster, cr); err != nil {
+		return err
+	}
+
+	if err := r.checkSuspendedDeadline(ctx, cluster, cr); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func checkStartingDeadline(ctx context.Context, cluster *api.PerconaXtraDBCluster, cr *api.PerconaXtraDBClusterBackup) error {
+	log := logf.FromContext(ctx)
+
+	if cr.Status.State != api.BackupNew {
+		return nil
+	}
+
+	var deadlineSeconds *int64
+	if cr.Spec.StartingDeadlineSeconds != nil {
+		deadlineSeconds = cr.Spec.StartingDeadlineSeconds
```

Comment:
Maybe it will be a bit nicer to use switch here?

Reply:
i don't think switch will be much different, why do you think that it might be better?

Reply:
```
var deadlineSeconds *int64

switch {
case cr.Spec.StartingDeadlineSeconds != nil:
    deadlineSeconds = cr.Spec.StartingDeadlineSeconds
case cluster.Spec.Backup.StartingDeadlineSeconds != nil:
    deadlineSeconds = cluster.Spec.Backup.StartingDeadlineSeconds
default:
    return nil
}
```
For me it looks more readable, but it's not critical for sure, just want to understand when what is better to use.